src/main/java/mate.academy/Main.java

```diff
@@ -4,13 +4,16 @@
 import mate.academy.lib.Injector;
 import mate.academy.model.Product;
 import mate.academy.service.ProductService;
+import mate.academy.service.impl.ProductServiceImpl;
 
 public class Main {
 
     public static void main(String[] args) {
         // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+        ProductService productService = (ProductService) injector
+                .getInstance(ProductServiceImpl.class);
```

Comment:
It seems that you're trying to cast the result of `getInstance` directly to `ProductService`. However, the `Injector` should be able to recognize that `ProductServiceImpl` implements `ProductService` and do the casting automatically if the setup is correct. Make sure that your `Injector` class is properly handling this.